### ISSUE TITLE
SIGSEGV (Segmentation fault) with high angular momentum (large zeta) basis sets

### ISSUE THREAD

**Issue by ndattani** [sunqm/qcint#17]
[Yesterday's update](https://github.com/sunqm/qcint/releases/tag/v5.1.7) helped me get past this error that I was getting when trying to do UHF for the O atom with aug-cc-pV7Z:
```
Program received signal SIGSEGV, Segmentation fault.
0x00007fffe755aa90 in CINTgout2e () from /cvmfs/[soft.computecanada.ca/easybuild/software/2020/avx2/Core/qcint/5.1.5/lib64/libcint.so.5](http://soft.computecanada.ca/easybuild/software/2020/avx2/Core/qcint/5.1.5/lib64/libcint.so.5)
```
Now I have a similar problem with aug-cc-pV8Z, but the problem is now with `CINTg4c_index_xyz` rather than `CINTgout2e`:
```
Program received signal SIGSEGV, Segmentation fault.
0x00007fffe753f117 in CINTg4c_index_xyz () from /cvmfs/[soft.computecanada.ca/easybuild/software/2020/avx2/Core/qcint/5.1.7/lib64/libcint.so.5](http://soft.computecanada.ca/easybuild/software/2020/avx2/Core/qcint/5.1.7/lib64/libcint.so.5)
```
The in-progress CFOUR code for high-z basis sets that I was working on debugging with Devin Matthews was able to converge energies up to 9Z but we couldn't compare it to anything due to limitations in MRCC. I guess I can use Gaussian16, OpenMolcas or Psi4 to help with this benchmark, but ideally PySCF could be included too [1]:
```
aug-cc-pV7Z:    -74.812392 7916845162 (MRCC,  268 functions)
aug-cc-pV7Z:    -74.812392 791677667  (CFOUR, 268 functions) 

aug-cc-pV8Z:    -74.812395 3518996672 (MRCC,  366 functions)
aug-cc-pV8Z:    -74.812395 351908577  (CFOUR, 366 functions, 110,596,892 2e- integrals) 

aug-cc-pV9Z:     MRCC: Error in Boys Function
aug-cc-pV9Z:    -74.812395 933100035  (CFOUR, 485 functions, 780,867,410 2e- integrals)
```
For H2, I was able to get 8Z and 9Z calculations done, but with 10Z, I got the same error that I got with 8Z for the oxygen atom:
```
Program received signal SIGSEGV, Segmentation fault.
0x00002aaabb33b117 in CINTg4c_index_xyz () from /cvmfs/soft.computecanada.ca/easybuild/software/2020/avx2/Core/qcint/5.1.7/lib64/libcint.so.5
```
Slightly different information is giving when I use `libcint` (same for O2 in aug-cc-pV8Z and aug-cc-pV10Z):
```
Program received signal SIGSEGV, Segmentation fault.
0x00002aaabb77fd9f in CINTg2e_index_xyz (idx=idx@entry=0x0, envs=envs@entry=0x7fffffff63e0)
    at /home/nike/wheels_builder/tmp.61293/python-3.10/pyscf-2.1.1/build/temp.linux-x86_64-cpython-310/deps/src/libcint/src/g2e.c:209
```

Related issues:

 - https://github.com/pyscf/pyscf/issues/685 (open since 19 August 2020)
 - https://github.com/pyscf/pyscf/issues/1181 (closed on 8 June 2022)
 - https://github.com/pyscf/pyscf/issues/1199 (closed on 8 June 2022)
 - https://github.com/sunqm/libcint/issues/86

[1] [Nike Dattani. (2021). HPQC-LABS/AI_ENERGIES (v2021.0) [Data set]. Zenodo. https://doi.org/10.5281/zenodo.5529103](https://github.com/HPQC-LABS/AI_ENERGIES/blob/master/high_angular_momentum_benchmarks.txt)

**Comment by ndattani** [comment]
This question has now also been [posted at MMSE](https://mattermodeling.stackexchange.com/q/10176/5) to see if others are getting the same issue.

**Comment by sunqm** [comment]
This problem is fixed in 5.1.8
